projects/angular-draggable-droppable/src/lib/droppable.directive.ts

```diff
@@ -47,6 +47,13 @@ export class DroppableDirective implements OnInit, OnDestroy {
    */
   @Input() dragActiveClass: string;
 
+  /**
+   * If set to true, will check the mouse event target to restrict dropping.
+   * This is useful if you would like to prevent dropping when there are floating
+   * elements (e.g. absolutely positioned) above the `mwlDroppable` element
+   */
+  @Input() restrictByEventTarget = false;
```

Comment:
Can we rework the api a little bit to make it a bit more flexible. I'm thinking something like we do on the draggable directive where we let the user [pass in a `validateDrag` function](https://github.com/cloud-maker-ai/angular-draggable-droppable/blob/feature/7293-restrict-target/projects/angular-draggable-droppable/src/lib/draggable.directive.ts#L129) that lets them control the behavior. 

So the API would be something like:
```
export interface ValidateDropParams {
  target: HtmlElement;
}

export type ValidateDrop = (params: ValidateDropParams) => boolean;

@Input() validateDrop: ValidateDrop;
```

and then this function can move into user land and you can have complete control over the behaviour in your own app:
```
private isTargetAllowed(target: EventTarget): boolean {
    if (!this.restrictByEventTarget) {
      return true;
    }

    const closestDroppableElement = (target as Element).closest(
      '[mwlDroppable]'
    );

    return closestDroppableElement === this.element.nativeElement;
  }
```

Reply:
Great idea! I'm happy to update it.

Do you think it's worth including are generic version of `isTargetAllowed` as an out-the-box drop validator?

e.g.
```ts
type DropValidator = (params: ValidateDropParams) => boolean;
interface DropValidators {
  restrictByEventTarget(selector: string) => DropValidator
}
```

```ts
import { DropValidators } from 'angular-draggable-droppable';
```

```html
<div [mwlDroppable] [validateDrop]="DropValidators.restrictByEventTarget('[mwlDroppable]')">
```

Reply:
Hey sorry about the late reply, I've been out on vacation. I think as the logic is pretty simple it's better to keep it in userland for now, it could always be added as a feature to the library in the future if there was a big demand for it

Reply:
Thanks @mattlewis92, that makes sense. I actaully started making the change the other day and almost immediately thought the same thing, not worth adding it here atm.

I have updated the change to expose the suggested `validateDrop` input. Please review when you have some time 😄

Reply:
Awesome, thank you! One last thing, could you add the 2 new interfaces here: https://github.com/mattlewis92/angular-draggable-droppable/blob/master/projects/angular-draggable-droppable/src/public_api.ts so that they can be used from the exported package as well

Reply:
done